languagetool-language-modules/fr/src/main/resources/org/languagetool/rules/fr/grammar.xml

```diff
@@ -37493,6 +37493,110 @@ though, and has since been slightly modified)-->
     </rule>
   </category>
   <category id="CAT_TOURS_CRITIQUES" name="Tours critiqués divers (barbarismes, impropriétés, solécismes, etc.)">
+      <rulegroup id="CONFUSION_DANS_UNE_EXPRESSION" name="confusion dans une expression" default="temp_off">
+          <rule>
+              <pattern>
+                  <marker>
+                      <token regexp="yes">(?-i)sans</token>
+                  </marker>
+                  <token>dessus</token>
+                  <token>dessous</token>
+              </pattern>
+              <message>Une erreur d’orthographe semble être détectée.</message>
+              <suggestion suppress_misspelled="yes"><match no="1" regexp_match="a" regexp_replace="e"/></suggestion>
+              <example correction="sens">Il a mis sa chambre <marker>sans</marker> dessus dessous pour retrouver son sac.</example>
+          </rule>
+          <rule>
+              <pattern>
+                  <token>je</token>
+                  <token>vous</token>
+                  <marker>
+                      <token regexp="yes">serais?</token>
+                  </marker>
+                  <token>gré</token>
+              </pattern>
+              <message>Une erreur d’orthographe semble être détectée.</message>
+              <suggestion suppress_misspelled="yes"><match no="3" regexp_match="serais?" regexp_replace="saurais"/></suggestion>
+              <example correction="saurais">Je vous <marker>serais</marker> gré de bien vouloir répondre à ce courrier.</example>
+          </rule>
+          <rule>
+              <antipattern>
+                  <token>de</token>
+                  <token>suite</token>
+                  <token regexp="yes">à|entières?</token>
+              </antipattern>
+              <antipattern>
+                  <token regexp="yes" skip="-1">pas|jamais|plus</token>
+                  <token>de</token>
+                  <token>suite</token>
+              </antipattern>
+              <antipattern>
+                  <token regexp="yes">toute?s?|ainsi|pas|jamais|plus|formes?|beaucoup|faute|défaites?</token>
```

Comment:
"trois fois de suite", is that a plausible expression? (if so, add "fois" to AP)

Reply:
Yes, that is possible so I added the AP.